lib/msf/core/modules/loader/base.rb

```diff
@@ -123,7 +128,7 @@ def load_module(parent_path, type, module_reference_name, options={})
 
     reload ||= force || file_changed
 
-    module_content = read_module_content(parent_path, type, module_reference_name)
+    module_content = read_module_content_from_path(module_path)
```

Comment:
Why does this method definition need to change? 👀

Reply:
`read_module_content_from_path(module_path)` has been added as a new method, `read_module_content(parent_path, type, module_reference_name)` still exists but nothing is calling it

The reason I added `read_module_content_from_path(module_path)` is because when it comes to payloads `module_path` (which is called by `read_module_content(parent_path, type, module_reference_name)` [here](https://github.com/rapid7/metasploit-framework/blob/master/lib/msf/core/modules/loader/directory.rb#L74)) doesn't produce the correct path since the layout of the payload modules differs from the rest of our modules

I did consider conditionally calling either  `read_module_content` or `read_module_content_from_path` but since I would need to update pro for the `read_module_content_from_path` workflow anyway I didn't see a benefit to it